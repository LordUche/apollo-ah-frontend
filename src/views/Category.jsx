import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Technology from '../../public/images/categories/technology.jpg';
import Sport from '../../public/images/categories/sports.jpg';
import Others from '../../public/images/categories/others.jpg';
import Fashion from '../../public/images/categories/fashion.jpg';
import Business from '../../public/images/categories/business.jpg';
import Food from '../../public/images/categories/food.jpg';
import Health from '../../public/images/categories/health.jpg';
import History from '../../public/images/categories/history.jpg';
import Entertainment from '../../public/images/categories/entertainment.jpg';


const Category = ({begin, end, articlesCategory}) => {
  return (
    <Fragment>
      <div className="container">
        <div className="row">
          {articlesCategory && articlesCategory.slice(begin, end).map(({ category }, index) => {
            return <div className="col-md-4 col-sm-6" key={index}>
              <Fragment>
                <div
                  className="category-item overlay category-item-technology"
                  style={{
                    backgroundImage: category === 'Technology' ? `url(${Technology})`:
                    category === 'Sports' ? `url(${Sport})` :
                    category === 'Fashion' ? `url(${Fashion})` :
                    category === 'Business' ? `url(${Business})` :
                    category === 'Health' ? `url(${Health})` :
                    category === 'Food' ? `url(${Food})` :
                    category === 'Entertainment' ? `url(${Entertainment})` :
                    category === 'History' ? `url(${History})` : `url(${Others})`
                  }}
                >
                <Link to="#" className="link-surface"/>
                  <div className="relative">
                    <h4>{category}</h4>
                    <span>Explore Articles</span>
                  </div>
                </div>
              </Fragment>
            </div>;
          })}
        </div>
      </div>
    </Fragment>
  );
};

Category.propTypes = {
  begin: PropTypes.number,
  end: PropTypes.number,
  articlesCategory: PropTypes.array.isRequired,
};

export default Category;
