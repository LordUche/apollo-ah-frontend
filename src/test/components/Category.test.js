/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import Category from '../../views/Category';

const props = [{ 
  begin: 0,
  end: 6,
  articlesCategory: [{
    id: 1,
    category: 'Technology',
  }, {
    id: 2,
    category: 'Business',
  }, {
    id: 3,
    category: 'Sports',
  }],
}];
describe('<Category Test Suite>', () => {
  describe('<Category>', () => {
    it('It should render succesfully', () => {
      const wrapper = shallow(<Category {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
