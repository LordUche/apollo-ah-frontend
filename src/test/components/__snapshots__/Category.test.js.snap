// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category Test Suite> <Category> It should render succesfully 1`] = `ShallowWrapper {}`;
